### PR TITLE
Feature marketplace

### PR DESCRIPTION
**Pull request by didntpay**
Add get list of item from organization and checkout items

### CODE REVIEW

File: src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                    .setItemName(item.getItemName()).setItemStock(item.getStock())
+                    .setBelongTo(item.getBelongTo())
+                    .setItemPrice(item.getItemPrice())
+                    .setItemDescription(item.getItemDescription())
+                    .setItemId(item.getItemID())
+                    .setItemImageLink(item.getItemImageLink())
+                    .build()
+            );
+        }
+
```
Comment (miska12345) [inline src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java:54]: Future improvement: Consider using grpc stream for returning a potentially long list of items.

File: src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        return GetOrgItemsResult.newBuilder().addAllItems(result).build();
+    }
+
+    public CheckOutResult checkout(String uesrId, CheckOutRequest request) {
+        List<ItemGrpc> items = request.getItemsList();
+        List<Integer> itemCounts = request.getCountList();
```
Comment (miska12345) [inline src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java:60]: Why is itemCounts separate from items?

File: src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+       Set<ItemGrpc> unprocessed = new HashSet<>(result.getUnprocessedItemList());
+       for (int i = 0; i < items.size(); i++) {
+           ItemGrpc item = items.get(i);
+           int count = itemCounts.get(i);
+
+           if (!unprocessed.contains(item)) {
+               total += this.itemDao.load(item.getItemId()).get().getItemPrice() * count;
+           }
+       }
+
```
Comment (miska12345) [inline src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java:85]: How can users know about these unprocessed items?

File: src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    private boolean checkTransactionStatus(String transactionId) {
+        for (int i = 0; i < CHECKOUT_TIMEOUT; i++) {
+            TransactionStatus paymentStatus = transactionServiceHandler.getTransactionStatus(transactionId);
+            if (paymentStatus.equals(TransactionStatus.COMPLETED)){
+                return true;
+            } else if (paymentStatus.equals(TransactionStatus.ERROR)) {
+                return false;
+            }
+            try{ Thread.sleep(2000); }
```
Comment (miska12345) [inline src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java:119]: Shouldn't this be 1000?

**Review by miska12345 — APPROVED** [review]

File: src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java (hunk 3 above)
Comment (didntpay) [reply src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java:84]: By writing them back to result

File: src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java (hunk 2 above)
Comment (didntpay) [reply src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java]: Fixed

File: src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java (hunk 4 above)
Comment (didntpay) [reply src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java:118]: Fixed in feature-getsimilar pull request

**Pull request merged**